replay-assets/replay_command_handlers.js

```diff
@@ -1725,12 +1725,133 @@ function getInternalFunctionLocationProp(cdpProperties) {
+    //    Commas can be part of initializers, but they would have to be contained
+    //    by complementary tokens, any of: ``""''[](){}. 
+    //    → Let's simply ignore everything inside of that.
+    // Special case 2: Argument destructuring.
+    //    Don't give it a name (empty string).
+    //    Same concept as case 1.
+
+    /**
+     * The function header without comments.
+     */
+    let header = "";
+    /**
+     * When in a comment, this is set to the counterpart that we are looking for.
+     * @type {string | null}
+     */
+    let expectedCommentEnd = null;
+    let parensStart = -1;
+    let insideInitializer = false;
+    /**
+     * @type {string[]}
+     */
+    const ignoreStack = [];
+    for (let i = 0; i < s.length; ++i) {
+      const c = s[i];
+      if (expectedCommentEnd) {
+        // In a comment.
+        if (hasSubstringAt(s, expectedCommentEnd, i)) {
+          // Comment End: Start new segment from here.
+          expectedCommentEnd = null;
+        } 
+      }
+      else {
+        // Not in a comment.
+        if (hasSubstringAt(s, "//", i) || hasSubstringAt(s, "/*", i)) {
+          // Comment Start.
+          if (c === "*") {
+            expectedCommentEnd = "*/";
+          } else {
+            expectedCommentEnd = "\n";
+          }
+          // Remove "/" from header:
+          header = header.slice(0, -1);
+        } else {
+          if (parensStart === -1) {
+            if (c === "(") {
+              // PARENS: Params start.
+              parensStart = header.length + 1;
+            } else if (hasSubstringAt(s, "=>", i)) {
+              // SINGLEARROW: Found the arrow → The last word in the header (sans "=") is the param.
+              const param = header.trim().match(/([^\s]+)\s*=$/)?.[1];
+              return param ? [param] : [];
+            }
+          } else if (c === "=") {
+            // Initializer start.
+            insideInitializer = true;
+          } else if (insideInitializer && c === ",") {
+            // Initializer end.
+            insideInitializer = false;
+          } else if (ignoreStack.length && ignoreStack[ignoreStack.length - 1] === c) {
+            // Inside destructuring argument or initializer expression:
+            // Exit node.
+            ignoreStack.pop();
+            continue; // Ignore this, too.
+          } else if (complementaryTokensStart.includes(c)) {
+            // Inside destructuring argument or initializer expression:
+            // Enter node.
+            const tokenIdx = complementaryTokensStart.indexOf(c);
+            ignoreStack.push(complementaryTokensEnd[tokenIdx]);
+          } else if (c === ")") {
+            // PARENS: Params end.
+            return header.substring(parensStart).split(",").map(p => p.trim());
+          }
+
+          if (!insideInitializer && !ignoreStack.length) {
+            // Add to the header.
+            header += c;
+          }
+        }
+      }
+    }
```

Comment:
I find this loop body really difficult to parse given many `if`/`elses` with different control flow in different branches (e.g. sometimes `continue`/`return`).

any chance I could get you to followup and reformulate it as a series of checks that _always_ `continue`/`break`/`return` and also remove the `else`s?  I imagine you could probably also copy the `header += c` into the various `then` branches and remove the extra `if` at the bottom.

Reply:
I'd love an SDT parser but hum...